src/responses/page-elements.ts

```diff
@@ -0,0 +1,79 @@
+import { Aggregation } from '../models/aggregation';
+
+/**
+ * The structure of the response body `hits` object returned from the PPS endpoint.
+ */
+export interface SearchResponseHits {
+  total: number;
+  returned: number;
+  hits: Record<string, unknown>[];
+}
+
+/**
+ * Valid page element names recognized & returned by the PPS
+ */
+export type PageElementName =
+  | 'uploads'
+  | 'reviews'
+  | 'collections'
+  | 'lending'
+  | 'web_archives'
+  | 'forum_posts';
+
+/**
+ * A map containing one or more page elements returned by the PPS, keyed by the
+ * name of the element.
+ */
+export interface PageElementMap
+  extends Partial<Record<PageElementName, unknown>> {
```

Comment:
small suggestion: move this under the declared types that are currently below it